integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioGateway.java

```diff
@@ -0,0 +1,460 @@
+    if (execResult.exitValue() == 0) {
+      logger.info("\n ===== HTTP response is \n " + execResult.stdout());
+      assertAll("Check that primary server name is not null or empty",
+          () -> assertNotNull(execResult.stdout(), "Primary server name shouldn’t be null"),
+          () -> assertFalse(execResult.stdout().isEmpty(), "Primary server name shouldn’t be  empty")
+      );
+
+      for (String httpAttrKey : httpAttrArray) {
+        String httpAttrValue = getHttpResponseAttribute(execResult.stdout(), httpAttrKey);
+        httpAttrInfo.put(httpAttrKey, httpAttrValue);
+      }
+    } else {
+      fail("==== Failed to process HTTP request " + execResult.stderr());
+    }
+
+    return httpAttrInfo;
+  }
+
+  private String buildCurlCommand(String curlUrlPath, String headerOption) {
+    final String httpHeaderFile = "/u01/domains/header";
+    final String clusterAddress = domainUid + "-cluster-" + clusterName;
+    logger.info("Build a curl command with pod name {0}, curl URL path {1} and HTTP header option {2}",
+        clusterAddress, curlUrlPath, headerOption);
+
+    int waittime = 5;
+    return new StringBuilder()
+        .append("curl --silent --show-error")
+        .append(" --connect-timeout ").append(waittime).append(" --max-time ").append(waittime)
+        .append(" http://")
+        .append(K8S_NODEPORT_HOST)
+        .append(":")
+        .append(istioIngressPort)
+        .append("/")
+        .append(curlUrlPath)
+        .append(headerOption)
+        .append(httpHeaderFile).toString();
+  }
+
+  private String getHttpResponseAttribute(String httpResponseString, String attribute) {
+    // retrieve the search pattern that matches the given HTTP data attribute
+    String attrPatn = httpAttrMap.get(attribute);
+    assertNotNull(attrPatn,"HTTP Attribute key shouldn’t be null");
+
+    // search the value of given HTTP data attribute
+    Pattern pattern = Pattern.compile(attrPatn);
+    Matcher matcher = pattern.matcher(httpResponseString);
+    String httpAttribute = null;
+
+    if (matcher.find()) {
+      httpAttribute = matcher.group(2);
+    }
+
+    return httpAttribute;
+  }
+
+  private void shutdownServerUsingServerStartPolicy(String msName) {
+    final String podName = domainUid + "-" + msName;
+
+    // shutdown a server by changing the it's serverStartPolicy property.
+    logger.info("Shutdown the server {0}", msName);
+    boolean serverStopped = assertDoesNotThrow(() ->
+        shutdownManagedServerUsingServerStartPolicy(domainUid, domainNamespace, msName));
+    assertTrue(serverStopped, String.format("Failed to shutdown server %s ", msName));
+
+    // check that the managed server pod shutdown successfylly
+    logger.info("Check that managed server pod {0} stopped in namespace {1}", podName, domainNamespace);
+    checkPodDoesNotExist(podName, domainUid, domainNamespace);
+  }
+
+  private Map<String, String> getServerAndSessionInfoAndVerify(String serverName,
```

Comment:
getServerAndSessionInfoAndVerify()  method is repeated in both class. Can be extracted out to a SessionMigration utility class to be used in both istio and non-istio usecases.

Reply:
changed